spec/solver_spec.rb

```diff
@@ -0,0 +1,30 @@
+require './solver'
+require 'rspec'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  describe '#factorial' do
+    it 'successfully implements the mathematical factorial method' do
+      expect(@solver.fizzbuzz(3)).to eq 'fizz'
+      expect(@solver.fizzbuzz(5)).to eq 'buzz'
+      expect(@solver.fizzbuzz(7)).to eq '7'
+      expect(@solver.factorial(-1)).to eq 'Wrong input, please provide positive integer numbers!'
+      expect(@solver.factorial(0)).to eq 1
+    end
+  end
+
+  describe '#reverse' do
+    it 'reverse the input string' do
+      expect(@solver.reverse('keenan')).to eq 'naneek'
+    end
+  end
+
+  describe '#fizzbuzz' do
+    it 'returns constructed string' do
+      expect(@solver.fizzbuzz(15)).to eq 'fizzbuzz'
+    end
+  end
```

Comment:
- [ ] As mentioned by the previous reviewer, kindly ensure that the `#fizzbuzz` method has also tests that handle the **[following aspects](https://github.com/keebot-png/TDD-project/pull/1#discussion_r1102551406)**:
  * Number is divisible by 3
  * Number is divisible by 5
  * Number is divisible by 3 and 5
  * Number is not divisible by 3
  * etc